File: Serene/Serene.Core/Modules/Northwind/NorthwindPermissionKeys.cs

using Serenity.Extensibility;
using System.ComponentModel;

namespace Serene.Northwind
{
    [NestedPermissionKeys]
    [DisplayName("Northwind")]
    public class PermissionKeys
    {
        [DisplayName("Customers")]
        public class Customer
        {
            public const string Delete = "Northwind:Customer:Delete";
            [Description("Create/Update")]
            public const string Modify = "Northwind:Customer:Modify";
            public const string View = "Northwind:Customer:View";
        }

        [Description("[General]")]
        public const string General = "Northwind:General";
    }
}

File: Serene/Serene.Core/Imports/ServerTypings/Northwind.PermissionKeys.ts
namespace Serene.Northwind {
    declare namespace PermissionKeys {
        export const General = "Northwind:General";

        namespace Customer {
            export const Delete = "Northwind:Customer:Delete";
            export const Modify = "Northwind:Customer:Modify";
            export const View = "Northwind:Customer:View";
        }
    }
}
